### ISSUE TITLE
Can't provide an empty hash as default value

### ISSUE THREAD

**Issue by science** [mattt/sinatra-param#37]
First off, great idea for a gem! This will be useful for me all over the place.

I was trying to create a param rule with a default value of an empty hash as:
```
param :category, Hash, default: {}
puts params.inspect #=> {}
```
So this doesn't seem to work - I get no default value ("category" remains undefined in params). What I would expect it to do is fill params as: `{"category" => {}}`

If I put a placeholder value in the hash, then I can get it to work, which is fine for me now, but it's ugly, and against the grain for this gem's design principle I think?
```
param :category, Hash, default: {nil: {}}
puts params.inspect #=> {"category"=>{:nil=>{}}}
```
Maybe I'm doing something wrong or misunderstanding something? Thanks again for this very useful idea!

**Comment by mattt** [comment]
Ah, I think I see what's going on: `{ }` is most likely being misinterpreted as a block, rather than an empty `Hash` in this context (for whatever reason). If you set `default` to `Hash.new`, everything should work as expected.

**Comment by ismaelga** [comment]
@mattt: If `{ }` is actually being interpreted as a block in this context it must be black magic of some sort. I strongly doubt that's the case here.

**Comment by mattt** [comment]
Meh. Chock it up to being in Swift-land these days.

Failing test added in 42cc688da44e3e597f563958f4a0e3946cd54ce8. Fix implemented in 5c1a3dff998ad1a0626d5b285464fc292e9517b8.

A new release is forthcoming. Thanks, @science!
